libnmstate/ifaces/ifaces.py

```diff
@@ -172,6 +172,7 @@ def _apply_copy_mac_from(self):
             if iface.type not in (
                 InterfaceType.LINUX_BRIDGE,
                 InterfaceType.BOND,
+                InterfaceType.OVS_INTERFACE,
```

Comment:
I think this will not work. We have to copy the MAC onto an internal interface which is modeled as a separate object in the "interfaces" list (unlike with bondings and bridges where the MAC is defined directly on the bridge/bond). Due to that, this will fail on line 182.

Reply:
Correct. The code should expand the check for OVS usage.

Reply:
Thanks! My intention (not sure it's the right one) was to add the copy-mac-from attribute to an ovs-interface, not to the ovs-bridge, so the usage would be for example:

```
    interfaces:
    - name: ovs0
      type: ovs-interface
      state: up
      ipv4:
        enabled: true
        dhcp: true
      copy-mac-from: eth0
    - name: ovs-br10
      type: ovs-bridge
      state: up
      bridge:
        options:
          stp: true
        port:
        - name: eth0
        - name: ovs0
```
Does this make sense or it should be added to the ovs-bridge? If that's the case, how do we know which internal interface to copy the mac to? IIUC? there may be many

Reply:
It should be added to ovs-interface. But to do that, you must extend the validation code of nmstate that makes sure that copy-mac-from references a port of the interface it is defined on, which is not the case with ovs-interface.